Hublsoft.Net.LoginLogout.DataAccess/RegisteredUsersRepository.cs

```diff
@@ -105,5 +105,65 @@ public async Task IncrementFailedLoginAttemptsAsync(int id)
                 await con.CloseAsync();
             }
         }
+
+        public async Task<int> GetFailedLoginAttempsAsync(int id)
+        {
+            if (id == 0)
+            {
+                throw new Exception("Id cannot be 0");
+            }
+
+            var attempCount = 0;
+
+            try
+            {
+                using (var con = new MySqlConnection(ConnectionString))
+                {
+                    await con.OpenAsync();
+
+                    var stm = $"SELECT FailedLoginAttempts FROM RegisteredUsers WHERE Id = {id} ;";
+
+                    using (var cmd = new MySqlCommand(stm, con))
+                    {
+                        attempCount = (int)cmd.ExecuteScalar();
+                    }
+
+                    await con.CloseAsync();
+                }
+            }
+            catch
+            {
+            }
+
+            return attempCount;
+        }
+
+        public async Task LockAccount(int id)
+        {
+            if (id == 0)
+            {
+                throw new ArgumentException(nameof(id));
+            }
+            using (var con = new MySqlConnection(ConnectionString))
+            {
+                await con.OpenAsync();
+
+                var stm = $"UPDATE RegisteredUsers SET Status = 1 WHERE Id = {id} ;";
+
+                using (var cmd = new MySqlCommand(stm, con))
+                {
+                    await cmd.ExecuteNonQueryAsync();
+                }
+
+                await con.CloseAsync();
```

Comment:
Closing a db connection will cause an exception when on line 164 the command is executed. It should be moved to after the using statement of the 2nd MySqlCommand.